test/BasicVaultTest.t.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import "forge-std/console2.sol";
+
+import 'src/BasicVault.sol';
+import 'test/WMDTokenWithFailedTransfers.sol';
+
+abstract contract StateZero is Test {
+        
+    BasicVault public vault;
+    WMDTokenWithFailedTransfers public wmd;
+    address user;
+
+    event Deposited(address indexed from, uint amount);
+    event Withdrawal(address indexed from, uint amount);
+
+    function setUp() public virtual {
+        wmd = new WMDTokenWithFailedTransfers();
+        vault = new BasicVault(wmd);
+
+        user = address(1);
+        vm.label(user, "user");
+    }
+}
+
+contract StateZeroTest is StateZero {
+    
+    function testGetUserBalance() public {
+        console2.log("Check User has zero tokens in vault");
+        vm.prank(user);
+        uint balance = vault.balances(user);
+        assertTrue(balance == 0);
+    }
+
+    function testUserCannotWithdraw(uint amount) public {
```

Comment:
we have been using the prefix `testFuzz` for fuzz tests. this makes it easier to identify and also allows us to easily exclude (or include) fuzz tests w regex from the command line with `forge test -m `